### PR TITLE
SPath: Implement more helper methods

### PR DESCRIPTION
**Pull request by LightArrowsEXE**
Implement some more methods that I would have uses for, and others may too.

### CODE REVIEW

**Comment by LightArrowsEXE** [comment]
`SPath.fglob` speeds

```bash
$ python benchmark.py 
Total time for 10000 iterations: 0.4242 seconds
Average time per fglob call: 0.000042 seconds
Total time for 10000 simple glob iterations: 0.5244 seconds
Average time per simple glob call: 0.000052 seconds
```

oh ye, we zoomin' now

File: stgpytools/types/file.py [hunk 1]
```diff
@@ -134,5 +138,58 @@ def move_dir(self, dst: SPath, *, mode: int = 0o777) -> None:
 
         self.rmdir()
 
+    def lglob(self, pattern: str = '*') -> list[SPath]:
+        """Glob the path and return the list of paths."""
+
+        return [SPath(p) for p in self.glob(pattern)]
```
Comment (Setsugennoao) [inline stgpytools/types/file.py]: list and map

File: stgpytools/types/file.py [hunk 2]
```diff
@@ -134,5 +138,58 @@ def move_dir(self, dst: SPath, *, mode: int = 0o777) -> None:
...
+        """Check if the directory is empty."""
+
+        return self.is_dir() and not any(self.iterdir())
+
+    def get_size(self) -> int:
+        """Get the size of the file or directory in bytes."""
+
+        if self.is_file():
+            return self.stat().st_size
+        elif self.is_dir():
```
Comment (Setsugennoao) [inline stgpytools/types/file.py]: remove elif, just if

File: stgpytools/types/file.py [hunk 3]
```diff
@@ -134,5 +138,58 @@ def move_dir(self, dst: SPath, *, mode: int = 0o777) -> None:
...
+            return sum(f.stat().st_size for f in self.rglob('*') if f.is_file())
```
Comment (Setsugennoao) [inline stgpytools/types/file.py]: so this only works for top-level files? does rglob used like this traverse dirs?

File: stgpytools/types/file.py [hunk 4]
```diff
@@ -134,5 +138,58 @@ def move_dir(self, dst: SPath, *, mode: int = 0o777) -> None:
...
+
+        raise ValueError("Path is neither a file nor a directory")
+
+    def backup(self, dst: SPath | None = None) -> SPath:
```
Comment (Setsugennoao) [inline stgpytools/types/file.py]: what's the use for this? and why not copy_dir?

File: stgpytools/types/file.py (hunk 3 above)
Comment (Vodes) [reply stgpytools/types/file.py]: What does the r stand for

Comment (Setsugennoao) [reply stgpytools/types/file.py]: 😁😃😄

**Pull request merged**